src/fdcache_fdinfo.cpp

```diff
@@ -159,8 +159,12 @@ bool PseudoFdInfo::GetEtaglist(etaglist_t& list)
     AutoLock auto_lock(&upload_list_lock);
 
     list.clear();
-    for(mppart_list_t::const_iterator iter = upload_list.begin(); iter != upload_list.end(); ++iter){
-        list.push_back(iter->etag);
+    for(filepart_list_t::const_iterator iter = upload_list.begin(); iter != upload_list.end(); ++iter){
+        if(iter->petag){
+            list.push_back(*(iter->petag));
+        }else{
+            S3FS_PRN_ERR("The pointer to the etag string is null(internal error). But will continue and the upload will fail.");
```

Comment:
Should s3fs eagerly fail this instead of waiting for the server to fail it after the upload?

Reply:
By returning an error with this method, I confirmed that EIO can immediately interrupt the process.
I have uploaded the code, so please check it.